### DISCUSSION TITLE
[Curve Editor] Snap tangent to hovered point.

### DISCUSSION THREAD

**Discussion by vvvvvvitor** [godotengine/godot-proposals#7218] [Editor]
The current way to make a tangent linear is quite hard to discover, I propose adding a snapping system to tangents
similar to the one on the Graph Editor.

Current:

[Gravação de tela de 2023-07-06 17-16-32.webm](https://github.com/godotengine/godot-proposals/assets/119557151/ab5111d5-414a-4b32-a110-1217819ffac2)

While holding a tangent, if you hover over a point, the tangent would automatically snap to it, making it essentially linear.
Ex.:

[Gravação de tela de 2023-07-06 17-18-24.webm](https://github.com/godotengine/godot-proposals/assets/119557151/5ca4c70b-1af2-42ea-b704-783ddfe64fac)

It's quite intuitive and efficient, just snap to a point and it's now linear, also, there would be no need for any extra buttons,
shortcuts or UI since it would all be done by the cursor.

**Comment by MewPurPur** [comment]
Good idea IMO, but should the tangent be effectively linear, or straight up become linear? IMO the latter is a better idea.

**Reply by vvvvvvitor** [reply]
I agree, the latter would probably be the best, maybe there could be some sort of indication, like a dotted like going to the point or something like that.

**Comment by Calinou** [comment]
@vvvvvvitor Remember that for GitHub video previews to work, you need to have a blank line before *and* after the video URL. I edited your post accordingly, but remember to do this in the future :slightly_smiling_face:

**Reply by vvvvvvitor** [reply]
Ah, I didn't know, thanks for fixing it!
